engine/src/main/java/org/camunda/bpm/engine/impl/util/CallableElementUtil.java

```diff
@@ -40,9 +44,38 @@ public static ProcessDefinitionImpl getProcessDefinitionToCall(VariableScope exe
     String processDefinitionKey = callableElement.getDefinitionKey(execution);
     String tenantId = callableElement.getDefinitionTenantId(execution);
 
+    return getCalledProcessDefinition(execution, callableElement, processDefinitionKey, tenantId);
+  }
+
+  public static ProcessDefinitionEntity getStaticallyBoundProcessDefinition(BaseCallableElement callableElement, String tenantId) {
+    if(callableElement.hasDynamicReferences()){
+      return null;
+    }
+
+    VariableScope emptyVariableScope = new StartProcessVariableScope();
+
+    if (callableElement.getTenantIdProvider() instanceof ElValueProvider) {
+      tenantId = callableElement.getDefinitionTenantId(emptyVariableScope);
+    }
+
+    String processDefinitionKey = callableElement.getDefinitionKey(emptyVariableScope);
+    try {
+      return getCalledProcessDefinition(emptyVariableScope, callableElement,
+        processDefinitionKey, tenantId);
+    } catch (NullValueException e) {
+      return null;
+    }
+  }
+
+  private static ProcessDefinitionEntity getCalledProcessDefinition(
+    VariableScope execution,
+    BaseCallableElement callableElement,
+    String processDefinitionKey,
+    String tenantId) {
+
     DeploymentCache deploymentCache = getDeploymentCache();
 
-    ProcessDefinitionImpl processDefinition = null;
+    ProcessDefinitionEntity processDefinition = null;
```

Comment:
Could you detail why this change is necessary?

Reply:
The original function`getProcessDefinitionToCall` from which this one was extracted returns `ProcessDefinitionImpl`, although it receives objects of type `ProcessDefinitionEntity` from the functions it calls. 
However, for its purpose the `Impl` type is sufficient, thats why it does not return the `Entity` type. In our case we need the `ProcessDefinintion` type to later check the access permission. `ProcessDefinitionEntity` both implements the `ProcessDefinintion ` interface and is a subtype of `ProcessDefinitionImpl` and is returned from the `DeploymentCache` methods anyways.

Reply:
I'll update `getStaticallyBoundProcessDefinition` to return `ProcessDefinition` as we certainly do not need to return the `Entity` type there.